File: MapGUI3/src/Model/IFileTable.java
package Model;

import java.io.BufferedReader;
import java.util.Map;

public interface IFileTable {
    boolean exists(String filename);
    int put(Map.Entry<String, BufferedReader> pair);
    Map.Entry<String, BufferedReader> get(int key);
    void remove(int fileDescriptor);
    BufferedReader getBufferReader(int fileDescriptor);
}

File: MapA7/MapA7/Model/IFileTable.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace MapA7.Model
{
    public interface IFileTable
    {
        int Add(Tuple<String, StreamReader> item);
        void Remove(int fd);
        bool Exists(String filename);
        StreamReader GetValue(int fd);
        void Clear();
    }
}
